src/Radicle/Internal/Core.hs

```diff
@@ -148,6 +150,23 @@ data ValueF r =
 
 instance Serialise r => Serialise (ValueF r)
 
+hashable :: (CPA t) => Annotated t ValueF -> Bool
+hashable = \case
+  PrimFn _ -> False
+  Ref _ -> False
+  Lambda{} -> False
+  List xs -> all hashable xs
+  Vec xs -> all hashable xs
+  Dict kvs -> getAll $ Map.foldMapWithKey (\k v -> All (hashable k && hashable v)) kvs
+  _ -> True
```

Comment:
Probably best to list all constructors explicitly. If a constructor is added we want the author to make an explicit decision here.